tests/integration_tests/tests/agentless_tests/test_blueprint_upload.py

```diff
@@ -0,0 +1,90 @@
+########
+# Copyright (c) 2020 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import requests
+import requests.status_codes
+
+from cloudify_rest_client.exceptions import CloudifyClientError
+
+from integration_tests import AgentlessTestCase
+from integration_tests.tests.utils import get_resource as resource
+
+
+class BlueprintValidateTest(AgentlessTestCase):
+    def test_blueprint_validate(self):
+        blueprint_id = 'bp-validate'
+        blueprint_filename = 'empty_blueprint.yaml'
+        self.client.blueprints.validate(
+            resource('dsl/{}'.format(blueprint_filename)),
+            entity_id=blueprint_id
+        )
+        temp_blueprint_id = self._assert_blueprint_validation_message(
+            blueprint_id, 'Blueprint validated.')
+        self._assert_cleanup(temp_blueprint_id)
+
+    def test_blueprint_validate_from_url(self):
+        blueprint_id = 'bp-url-validate'
+        archive_url = 'https://cloudify-tests-files.s3-eu-west-1.amazonaws' \
+                      '.com/blueprints/the-not-blueprint-master.zip'
+        self.client.blueprints.validate(
+            archive_url,
+            blueprint_id)
+        temp_blueprint_id = self._assert_blueprint_validation_message(
+            blueprint_id, 'Blueprint validated.')
+        self._assert_cleanup(temp_blueprint_id)
+
+    def test_blueprint_validate_invalid_blueprint(self):
+        blueprint_id = 'bp-bad-schema'
```

Comment:
This is the same flow as here https://github.com/cloudify-cosmo/cloudify-manager/pull/2533/files#diff-6a28a93db36b5a4785a60c49c9b49d01a993f8f666d75a5f82c2e7ec015a4afdR27

You can take it out to a function.

Reply:
done